docs/Location.md

```diff
@@ -12,6 +12,23 @@ key           A unique identifier for this location
 
 Support for query string parsing is provided using the [`useQueries` module](QuerySupport.md).
 
+### Location Specs
+
+[Location specs](Glossary.md#location) can be either objects or path strings. As objects, they are like `location` objects, except they ignore the internally-generated `action` and `key` fields.
+
+You can use location specs to call `history.push` and `history.replace`. Location spec objects can define just the portions of the next `location` that you want to set. They can also extend an existing `location` object to change only specific fields on that `location`.
+
+```js
+// Pushing a path string.
+history.push('/the/path')
+
+// Omitting location state when pushing a location spec.
+history.push({ pathname: '/the/path', search: '?the=search' })
+
+// Extending an existing location object.
+history.push({ ...location, search: '?other=search' })
+```
+
 ### Programmatic Creation
```

Comment:
Just a side note- hopefully we can get rid of programmatic creation of location objects now that we have the concept of a location spec. I'd prefer to not have too much API for doing this. Haven't thought it all the way through yet though.